nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/ColdStorageHelper.java

```diff
@@ -48,6 +54,12 @@
 
     public static final String COLD_STORAGE_BEING_RETRIEVED_PROPERTY = "coldstorage:beingRetrieved";
 
+    public static final String GET_DOCUMENTS_TO_CHECK_QUERY = String.format(
+            "SELECT * FROM Document WHERE %s = 1 and %s = 0", COLD_STORAGE_BEING_RETRIEVED_PROPERTY,
+            NXQL.ECM_ISTRASHED);
```

Comment:
Do we also want to filter versions?

Reply:
good question but i don't really hava an answer :) but i am wondering if the check on `beingretrieved` is good enough and sufficient on `S3` moving to `Glacier` will create a new version but i don't know exactly if there is an effect on the query

Reply:
@efge WDYT?

Reply:
I don't see why you would filter on anything, it's not this query's job to decide where coldstorage can be used. You should do `SELECT * FROM Document, Relation WHERE coldstorage:beingRetrieved = 1` and that's it. The presence of the index on `coldstorage:beingRetrieved` will make sure this is fast anyway.